docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+
+The application you are building today is a Slash command Slack App to submit and manage general "work" requests in a Slack channel.
+
+### System architecture
+
+**What are the components of the system and how are they arranged?**
+
+The system consists of a Slack Workspace, a Slack App, a Temporal Service, and a Temporal Application.
+
+Traditionally, you would need a database to persist the state of your Slack App between interactions in case the Slack App process crashes or goes down.
+If you use Temporal, you can persist the state of your Slack App directly within a Temporal Application.
+
+![System component architecture](./images/temporal-slack-app.svg)
+
+The preceding diagram abstracts Slack's back end as a communication channel between the Slack App and the Slack Workspace.
+The diagram could also abstract the Temporal Service as a communication channel.
+However, for folks learning Temporal, it's helpful to understand that the Temporal Service is a separate service from the Temporal Application.
+
+**Is trading off a database for Temporal worth it?**
+
+Setting up and managing your database can be more overhead.
+During the development process instead of a managing a database, you will run a Temporal Service locally alongside your Temporal Application.
+You should already have it running if you completed the [Prerequisite steps](#prerequisites).
+So, during development it will seem like you're managing two things instead of one.
+
+However, after the development process, if you use Temporal Cloud as your Temporal Service, you only need to deploy the Temporal Application the same way as you would your Slack App.
+Then, your Temporal Application replaces an entire database with less than 70 lines of code.
+
+![Temporal development vs production](./images/temporal-slack-app-dev-vs-prod.svg)
+
+To use Temporal Cloud in production, see the [Deploy your TypeScript application to Digital Ocean](/tutorials/typescript/digital-ocean-deployment) tutorial for more information.
+
+### Functional requirements
+
+**What are the functional requirements of the application?**
+
+1. Using a Slash command, a user can submit a work request.
+2. Using a Slash command, a user can see all the work requests for the channel.
+3. When viewing all the work requests for the channel, a user can click a button to claim a work request.
+4. When viewing all the work requests for the channel, a user can click a button to mark a work request as complete.
+5. Using a Slash command, a user can delete the Work Queue for the channel.
+
+![App functionality demo](./images/work-queue-demo.gif)
+
+## Create a new Slack App configuration
+
+Go to https://api.slack.com/apps?new_app=1 and create an app "From scratch." Name the app whatever you want, and select a workspace that you intend to install it to.
+
+Once you've done this, click "Create App."
+
+### Enable Socket Mode
```

Comment:
Doesn't need to be an h3 (or even a new paragraph) imo -- I would just start the following sentence with "Next, under "Settings" in the navigation bar..."

Reply:
headers create anchors - anchors are important to enable deep linking directly to places in content.

I don't think we should shy away from headers just because Docusuarus TOC is jumbly - obviously not great, but if we switch away from Docusaurus we would want these headers / anchors here.